packages/adminrouter/extra/src/lib/util.lua

```diff
@@ -56,12 +56,116 @@ end
 
 
 function util.table_len(tbl)
+    -- Return the length of the table, without the limitations of #tbl
+    -- approach, where nil value in the table terminates counting/acts as the
+    -- end of the table.
     local count = 0
     for _ in pairs(tbl) do count = count + 1 end
     return count
 end
 
 
+function util.reverse(tbl)
+    -- Reverse in-place given table
+    local tl = util.table_len(tbl)
+
+    for i=1, math.floor(tl / 2) do
+        local tmp = tbl[i]
+        tbl[i] = tbl[tl - i + 1]
+        tbl[tl - i + 1] = tmp
+    end
+end
+
+
+function util.extract_service_path_component(service_path, fieldsLimit)
+    -- Extract path component in normalized form, at given level.
+    --
+    -- This function extracts from given service path the normalized service id
+    -- at a given level, defined by fieldsLimit.For example:
+    --
+    -- service_path group1/jenkins/ver/important/path/foobar.js
+    --
+    -- will yield following results, depending on fieldsLimit parameter:
+    -- nil:
+    --   normalised_name: foobar.js.path.important.ver.jenkins.group1
+    --   plain_name: group1/jenkins/ver/important/path/foobar.js
+    --   noreSegments: false
+    -- 4:
+    --   normalised_name: important.ver.jenkins.group1
+    --   plain_name: group1/jenkins/ver/important
+    --   noreSegments: true
+    -- 3:
+    --   normalised_name: ver.jenkins.group1
+    --   plain_name: group1/jenkins/ver
+    --   noreSegments: true
+    -- 2:
+    --   normalised_name: jenkins.group1
+    --   plain_name: group1/jenkins
+    --   noreSegments: true
+    -- 1:
+    --   normalised_name: group1
+    --   plain_name: group1
+    --   noreSegments: true
+    --
+    -- Returns:
+    --   A list with following elements, in order:
+    --     - normalised_name: service name in normalized form
+    --     - plain_name: service name in plain form
+    --     - moreSegments: true/false depending on whether all segments
+    --       service_path were processed or not.
+    local tmpTbl = {}
+    local moreSegments = false
+
+    if service_path:len() > 0 then
+        fieldsLimit = fieldsLimit or -1
+
+        local fieldCursor, searchCursor = 1, 1
+        local substrStart, substrEnd = service_path:find('/', searchCursor, true)
```

Comment:
I believe as you're searching for a single-character string, `substrStart === substrEnd`. Perhaps we should use that fact and do
```
sepIdx, _ = service_path.find('/', ...)
```